sklearn/tests/test_calibration.py

```diff
@@ -182,6 +188,64 @@ def test_sample_weight(data, method, ensemble):
     assert diff > 0.1
 
 
+@pytest.mark.parametrize("method", ["sigmoid", "isotonic"])
+@pytest.mark.parametrize("ensemble", [True, False])
+def test_sample_weight_class_imbalanced(method, ensemble):
+    X, y = make_blobs((100, 1000), center_box=(-1, 1), random_state=42)
+
+    # Compute weigths to compensate the unbalance of the dataset
+    sample_weight = 9 * (y == 0) + 1
+
+    X_train, X_test, y_train, y_test, sw_train, sw_test = train_test_split(
+        X, y, sample_weight, stratify=y, random_state=42
+    )
+
+    scaler = StandardScaler()
```

Comment:
We have a more general silent bug in scikit-learn while passing `Pipeline` where `sample_weight` will be ignored by an estimator like the calibration estimator (or the bagging meta-estimator). 

However, this bug is not anymore in the scope of this PR and will require some other changes in scikit-learn, therefore, I would propose to add a comment and fix only the initial bug. I will open a subsequent issue to discuss the potential solutions.

```suggestion
    # FIXME: ideally we should create a `Pipeline` with the `StandardScaler`
    # followed by the `LinearSVC`. However, `Pipeline` does not expose
    # `sample_weight` and it will be silently ignored.
    scaler = StandardScaler()
```